build.sbt

```diff
@@ -1496,6 +1497,7 @@ lazy val lmCoursierShaded = project
     Mima.lmCoursierFilters,
     Mima.lmCoursierShadedFilters,
     Compile / sources := (lmCoursier / Compile / sources).value,
+    Compile / scalacOptions --= Seq("-Xfatal-warnings"),
```

Comment:
Needs this to workaround

```
[error] -- Error: /home/runner/work/sbt/sbt/lm-coursier/src/main/scala/lmcoursier/CoursierDependencyResolution.scala:58:14 
[error] 58 |      Try(new URL(rawURL)) match {
[error]    |              ^^^
[error]    |constructor URL in class URL is deprecated since : see corresponding Javadoc for more information.
```

https://github.com/coursier/sbt-coursier/pull/526 is made to fix the root cause, we can remove the workaround after next sbt-coursier version is released.